crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -382,6 +390,157 @@ impl Persistence {
         Ok(solution)
     }
 
+    pub async fn solvable_order_after(
+        &self,
+        current_orders: &boundary::SolvableOrders,
+        after_timestamp: DateTime<Utc>,
+        after_block: u64,
+        min_valid_to: u32,
+    ) -> anyhow::Result<boundary::SolvableOrders> {
+        let after_block = i64::try_from(after_block).context("block number value exceeds i64")?;
+        let mut tx = self.postgres.pool.begin().await.context("begin")?;
+        // Set the transaction isolation level to REPEATABLE READ
+        // so the both SELECT queries below are executed in the same database snapshot
+        // taken at the moment before the first query is executed.
+        sqlx::query("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ")
+            .execute(tx.deref_mut())
+            .await?;
+
+        // Find order uids for orders that were updated after the given block.
+        let updated_order_uids: Vec<database::OrderUid> = {
+            let _timer = Metrics::get()
+                .database_queries
+                .with_label_values(&["updated_order_uids"])
+                .start_timer();
+
+            database::orders::updated_order_uids_after(&mut tx, after_block)
+                .map_ok(|uid| uid.0)
+                .try_collect()
+                .await?
+        };
+
+        // Fetch the orders that were updated after the given block and were created or
+        // cancelled after the given timestamp.
+        let next_orders: HashMap<domain::OrderUid, model::order::Order> = {
+            let _timer = Metrics::get()
+                .database_queries
+                .with_label_values(&["open_orders_after"])
+                .start_timer();
+
+            database::orders::open_orders_after(&mut tx, &updated_order_uids, after_timestamp)
+                .map(|result| match result {
+                    Ok(order) => full_order_into_model_order(order)
+                        .map(|order| (domain::OrderUid(order.metadata.uid.0), order)),
+                    Err(err) => Err(anyhow::Error::from(err)),
+                })
+                .try_collect()
+                .await?
+        };
+
+        let next_order_uids = next_orders.keys().cloned().collect::<HashSet<_>>();
+
+        // Interactions table doesn't contain block number, so we need to update
+        // interactions for orders from the cache.
+        let current_order_interactions = {
```

Comment:
`current_order_interactions` and `updated_quotes` could be parallelized.

Reply:
ditto: https://github.com/cowprotocol/services/pull/2923#discussion_r1740655235